ntbs-service/Services/NotificationService.cs

```diff
@@ -265,13 +267,28 @@ private async Task UpdateDatabase(AuditType auditType = AuditType.Edit)
         }
 
         public IQueryable<Notification> FilterById(IQueryable<Notification> IQ, string IdFilter) {
-            return IQ.Where(s => s.NotificationId.Equals(Int32.Parse(IdFilter)) 
+            int parsedIdFilter;
+            int.TryParse(IdFilter, out parsedIdFilter);
+            return IQ.Where(s => s.NotificationId.Equals(parsedIdFilter) 
```

Comment:
small thing - the `I` in the variable names seems a bit misleading - on types it indicates the interface, but here I think we could just say `queryable` (or just `notifications`?)